crates/sui-core/src/consensus_handler.rs

```diff
@@ -53,11 +62,75 @@ impl<T> ConsensusHandler<T> {
             checkpoint_service,
             transaction_manager,
             parent_sync_store,
+            low_scoring_authorities: scores_per_authority,
+            committee,
             metrics,
         }
     }
 }
 
+/// Updates list of authorities that are deemed to have low reputation scores by consensus
+/// these may be lagging behind the network, byzantine, or not reliably participating for any reason.
+/// We want to ensure that the remaining set of validators once we exclude the low scoring authorities
+/// is including enough stake for a quorum, at the very least. It is also possible that no authorities
+/// are particularly low scoring, in which case this will result in storing an empty list.
+fn update_low_scoring_authorities(
+    low_scoring_authorities: Arc<DashMap<AuthorityName, u64>>,
+    committee: Committee,
+    reputation_scores: ReputationScores,
+) {
+    if !reputation_scores.final_of_schedule {
+        return;
+    }
+    let mut score_list = vec![];
+    for val in reputation_scores.scores_per_authority.values() {
+        score_list.push(*val as f64);
+    }
+
+    let median = statistical::median(&score_list);
+    let mut deviations = vec![];
+    let mut abs_deviations = vec![];
+    for (i, _) in score_list.clone().iter().enumerate() {
+        deviations.push((score_list[i] - median) * -1.0);
+        abs_deviations.push((score_list[i] - median).abs());
+    }
+
+    // adjusted median absolute deviation
+    let mad = statistical::median(&abs_deviations) / 0.5;
```

Comment:
Why the division by `0.5` ?

Reply:
Ok looking on this I understand that you want to use the media absolute deviation to do an outlier detection in our sampling, but could you please explain/cite the approach you used to calculate this and also the choice of constants (ex `0.5` , `2.5` ) it would really help so can better understand the intention here.

Reply:
We can't exclude anyone here because the list position indicates the authority it corresponds to.
As for the -1, we only want to remove low value outliers, not high value outliers. I have updated it to not multiply by -1 and check if < - 2.4 instead, which is equivalent.
I'll add more comments